src/order/filter/at_top_bot.lean

```diff
@@ -685,6 +685,77 @@ lemma map_at_bot_eq_of_gc [semilattice_inf α] [semilattice_inf β] {f : α →
   map f at_bot = at_bot :=
 @map_at_top_eq_of_gc (order_dual α) (order_dual β) _ _ _ _ _ hf.order_dual gc hgi
 
+lemma map_coe_at_top_of_Ici_subset [semilattice_sup α] {a : α} {s : set α} (h : Ici a ⊆ s) :
+  map (coe : s → α) at_top = at_top :=
+begin
+  have : directed (≥) (λ x : s, 𝓟 (Ici x)),
+  { intros x y,
+    use ⟨x ⊔ y ⊔ a, h le_sup_right⟩,
+    simp only [ge_iff_le, principal_mono, Ici_subset_Ici, ← subtype.coe_le_coe, subtype.coe_mk],
+    exact ⟨le_sup_left.trans le_sup_left, le_sup_right.trans le_sup_left⟩ },
+  haveI : nonempty s := ⟨⟨a, h le_rfl⟩⟩,
+  simp only [le_antisymm_iff, at_top, le_infi_iff, le_principal_iff, mem_map, mem_set_of_eq,
+    map_infi_eq this, map_principal],
+  split,
+  { intro x,
+    refine mem_sets_of_superset (mem_infi_sets ⟨x ⊔ a, h le_sup_right⟩ (mem_principal_self _)) _,
+    rintro _ ⟨y, hy, rfl⟩,
+    exact le_trans le_sup_left (subtype.coe_le_coe.2 hy) },
+  { intro x,
+    filter_upwards [mem_at_top (↑x ⊔ a)],
+    intros b hb,
+    exact ⟨⟨b, h $ le_sup_right.trans hb⟩, subtype.coe_le_coe.1 (le_sup_left.trans hb), rfl⟩ }
+end
+
+/-- The image of the filter `at_top` on `Ici a` under the coercion equals `at_top`. -/
+@[simp] lemma map_coe_Ici_at_top [semilattice_sup α] (a : α) :
+  map (coe : Ici a → α) at_top = at_top :=
+map_coe_at_top_of_Ici_subset (subset.refl _)
+
+/-- The image of the filter `at_top` on `Ioi a` under the coercion equals `at_top`. -/
+@[simp] lemma map_coe_Ioi_at_top [semilattice_sup α] [no_top_order α] (a : α) :
+  map (coe : Ioi a → α) at_top = at_top :=
+begin
+  rcases no_top a with ⟨b, hb⟩,
+  exact map_coe_at_top_of_Ici_subset (Ici_subset_Ioi.2 hb)
+end
+
+/-- The `at_top` filter for an open interval `Ioi a` comes from the `at_top` filter in the ambient
+order. -/
+lemma at_top_Ioi_eq [semilattice_sup α] [no_top_order α] (a : α) :
+  at_top = comap (coe : Ioi a → α) at_top :=
+by rw [← map_coe_Ioi_at_top a, comap_map subtype.coe_injective]
+
+/-- The `at_top` filter for an open interval `Ici a` comes from the `at_top` filter in the ambient
+order. -/
+lemma at_top_Ici_eq [semilattice_sup α] (a : α) :
```

Comment:
My intuition would flip the sides of these lemmas and also make them simp. But presumably you have a reason for stating them like this?

Reply:
My reason to add this lemma is to replace `tendsto f _ (at_top : filter (Ici a))` with `tendsto (coe ∘ f) _ at_top`. I assume that `at_top : filter α` is simpler than `at_top : filter (Ici a)`, so the simplification goes this way.